src/ray/raylet/scheduling/scheduling_policy.h

```diff
@@ -59,28 +64,24 @@ class SchedulingPolicy {
   /// will act like a traditional weighted round robin.
   ///
   /// \param resource_request: The resource request we're attempting to schedule.
-  /// \param scheduler_avoid_gpu_nodes: if set, we would try scheduling
-  /// CPU-only requests on CPU-only nodes, and will fallback to scheduling on GPU nodes if
-  /// needed.
+  /// \param scheduling_options: scheduling options.
   ///
-  /// \return -1 if the task is unfeasible, otherwise the node id (key in `nodes`) to
-  /// schedule on.
-  scheduling::NodeID HybridPolicy(
-      const ResourceRequest &resource_request, float spread_threshold,
-      bool force_spillback, bool require_available,
-      bool scheduler_avoid_gpu_nodes = RayConfig::instance().scheduler_avoid_gpu_nodes());
+  /// \return NodeID::Nil() if the task is unfeasible, otherwise the node id
+  /// to schedule on.
+  scheduling::NodeID HybridPolicy(const ResourceRequest &resource_request,
```

Comment:
How about to make each policy a single class?

Reply:
Chatted with @wumuzi520 offline. The current thinking is we can make SchedulingPolicy an interface, and we could implement SchedulingPolicy interface with individual policy or a union of multiple policies. These should all hide behind SchedulingPolicy interfaces.